homeassistant/components/lg_netcast/config_flow.py

```diff
@@ -0,0 +1,216 @@
+"""Config flow to configure the LG Netcast TV integration."""
+from __future__ import annotations
+
+import contextlib
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING, Any
+
+from pylgnetcast import AccessTokenError, LgNetCastClient, SessionIdError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_HOST,
+    CONF_ID,
+    CONF_MODEL,
+    CONF_NAME,
+)
+from homeassistant.core import CALLBACK_TYPE, DOMAIN as HOMEASSISTANT_DOMAIN, callback
+from homeassistant.data_entry_flow import AbortFlow
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+from homeassistant.util.network import is_host_valid
+
+from .const import DEFAULT_NAME, DOMAIN
+from .helpers import LGNetCastDetailDiscoveryError, async_discover_netcast_details
+
+DISPLAY_ACCESS_TOKEN_INTERVAL = timedelta(seconds=1)
+
+
+class LGNetCast(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for LG Netcast TV integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.client: LgNetCastClient | None = None
+        self.device_config: dict[str, Any] = {}
+        self._discovered_devices: dict[str, Any] = {}
+        self._track_interval: CALLBACK_TYPE | None = None
+
+    def create_client(self) -> None:
+        """Create LG Netcast client from config."""
+        host = self.device_config[CONF_HOST]
+        access_token = self.device_config.get(CONF_ACCESS_TOKEN)
+        self.client = LgNetCastClient(host, access_token)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            if is_host_valid(host):
+                self.device_config[CONF_HOST] = host
+                return await self.async_step_authorize()
+
+            errors[CONF_HOST] = "invalid_host"
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_HOST): str}),
+            errors=errors,
+        )
+
+    async def async_step_import(self, config: dict[str, Any]) -> ConfigFlowResult:
+        """Import configuration from yaml."""
+        self.device_config = {
+            CONF_HOST: config[CONF_HOST],
+            CONF_NAME: config[CONF_NAME],
+        }
+
+        def _create_issue():
+            async_create_issue(
+                self.hass,
+                HOMEASSISTANT_DOMAIN,
+                f"deprecated_yaml_{DOMAIN}",
+                breaks_in_ha_version="2024.6.0",
```

Comment:
This needs to be bumped now
```suggestion
                breaks_in_ha_version="2024.11.0",
```